cli/tests/pcluster/config/test_config_patch.py

```diff
@@ -0,0 +1,266 @@
+    dst_conf = PclusterConfig(config_file=rendered_config_file,)
+
+    change = Change(
+        section_key, section_label, param_key, src_param_value, dst_param_value, Updatability(change_updatability_level)
+    )
+    check_patch(src_conf, dst_conf, [change], patch_applicability)
+
+
+def test_multiple_param_changes(mocker, pcluster_config_reader, test_datadir):
+    mocker.patch("pcluster.config.param_types.get_avail_zone", return_value="mocked_avail_zone")
+    dst_config_file = "pcluster.config.dst.ini"
+    duplicate_config_file(dst_config_file, test_datadir)
+
+    src_dict = {}
+    src_dict.update(defaults)
+    src_dict["master_subnet_id"] = "subnet-12345678"
+    src_dict["compute_subnet_id"] = "subnet-12345678"
+    src_dict["additional_sg"] = "sg-12345678"
+
+    rendered_config_file = pcluster_config_reader(**src_dict)
+    src_conf = PclusterConfig(config_file=rendered_config_file, fail_on_file_absence=True)
+
+    dst_dict = {}
+    dst_dict.update(defaults)
+    dst_dict["master_subnet_id"] = "subnet-1234567a"
+    dst_dict["compute_subnet_id"] = "subnet-1234567a"
+    dst_dict["additional_sg"] = "sg-1234567a"
+
+    rendered_config_file = pcluster_config_reader(dst_config_file, **dst_dict)
+    dst_conf = PclusterConfig(config_file=rendered_config_file,)
+
+    check_patch(
+        src_conf,
+        dst_conf,
+        [
+            Change(
+                "vpc", "default", "master_subnet_id", "subnet-12345678", "subnet-1234567a", Updatability(Updl.DENIED)
+            ),
+            Change(
+                "vpc", "default", "compute_subnet_id", "subnet-12345678", "subnet-1234567a", Updatability(Updl.DENIED)
+            ),
+            Change("vpc", "default", "additional_sg", "sg-12345678", "sg-1234567a", Updatability(Updl.ALLOWED)),
+        ],
+        Updl.DENIED,
+    )
+
+
+def _test_equal_configs(base_conf, target_conf):
+    # Without doing any changes the two configs must be equal
+    check_patch(base_conf, target_conf, [], Updl.ALLOWED)
+
+
+def _test_less_target_sections(base_conf, target_conf):
+    # Remove an ebs section in the target conf
+    assert target_conf.get_section("ebs", "ebs-1") is not None
+    target_conf.remove_section("ebs", "ebs-1")
+    assert target_conf.get_section("ebs", "ebs-1") is None
+
+    # The patch must show 2 differences: one for ebs_settings and one for missing ebs section in target conf
+    check_patch(
+        base_conf,
+        target_conf,
+        [
+            Change(
+                "cluster",
+                "default",
+                "ebs_settings",
+                "_ebs,ebs-2",
+                "ebs-2",
+                Updatability(Updl.DENIED, fail_reason="Cannot add/remove EBS Sections"),
+            ),
+            Change("ebs", "_ebs", "shared_dir", "vol1", None, Updatability(Updl.DENIED)),
+        ],
+        Updl.DENIED,
+    )
+
+
+def _test_more_target_sections(base_conf, target_conf):
```

Comment:
probably these three tests can be merged with a parametrize function

Reply:
Not sure about it. I'm adding, removing or replacing EBS sections before running the checks. I think it would become less explicit what we are testing here, even if possible.